File: LC-1512/LC-1512.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
  let count = {}
  for (let i = 0; i < nums.length; ++i) {
    count[nums[i]] = (count[nums[i]] || 0) + 1
  }

  let pairCount = 0
  for (let [key, val] of Object.entries(count)) {
    if (val > 2) {
      pairCount += (val * (val - 1)) / 2
    } else if (val === 2) {
      pairCount += 1
    }
  }
  return pairCount
}

File: LC-1512/LC-1512.cpp
#include <unordered_map>

using namespace std;

class Solution {
public:
    /**
     * Concepts: Use ordered-map to count each identical number in array
     */
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int> bucket;
        for (int i = 0; i < nums.size(); ++i) {
            ++bucket[nums[i]];
        }
        
        int pairCount = 0;
        for (int i = 0; i < bucket.size(); ++i) {
            if (bucket[i] > 2)
                pairCount += (bucket[i] * (bucket[i] - 1)) / 2;
            else if (bucket[i] == 2)
                ++pairCount;
        }
        return pairCount;
    }
};